protocols/kad/src/handler.rs

```diff
@@ -90,7 +91,13 @@ pub struct KademliaHandler<TUserData> {
     next_connec_unique_id: UniqueConnecId,
 
     /// List of active outbound substreams with the state they are in.
-    outbound_substreams: SelectAll<OutboundSubstreamState<TUserData>>,
+    active_outbound_substreams: SelectAll<OutboundSubstreamState<TUserData>>,
+
+    /// Number of outbound streams being upgraded right now.
+    num_upgrading_outbound_streams: usize,
+
+    /// List of outbound substreams that are waiting to become active next.
+    next_outbound_streams: VecDeque<OutboundSubstreamState<TUserData>>,
```

Comment:
Do I understand correctly that `next_outbound_streams` would only ever contain an `OutboundSubstreamState::PendingOpen`? If so, what do you think of not reusing the same `enum` as in `active_outbound_substreams`, but instead split the `enum` in two?